src/script_opt/Expr.cc

```diff
@@ -2119,7 +2119,12 @@ ExprPtr ArithCoerceExpr::Reduce(Reducer* c, StmtPtr& red_stmt)
 		const auto& ct = cv->GetType();
 
 		if ( IsArithmetic(t->Tag()) || IsArithmetic(ct->Tag()) )
-			return make_intrusive<ConstExpr>(FoldSingleVal(cv, t));
+			{
+			auto v = FoldSingleVal(cv, t);
```

Comment:
Style nit: this can be `if ( auto v = FoldSingleVal(cv, t) )`

Reply:
That idiom still feels a little funny to me :-), so I'm apt to miss these, but I've changed that one.